ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -1659,22 +1902,55 @@ pub(super) struct X64CompiledTrace {
     gdb_ctx: GdbCtx,
 }
 
+impl X64CompiledTrace {
+    /// Return the locations of the live variables this trace expects as input.
+    fn entry_vars(&self) -> &[VarLocation] {
+        &self.entry_vars
+    }
+}
+
 impl CompiledTrace for X64CompiledTrace {
     fn entry(&self) -> *const libc::c_void {
         self.buf.ptr(AssemblyOffset(0)) as *const libc::c_void
     }
 
     fn sidetraceinfo(&self, gidx: GuardIdx) -> Arc<dyn SideTraceInfo> {
-        // FIXME: Can we reference these instead of copying them?
-        let aotlives = self.deoptinfo[&usize::from(gidx)]
+        // FIXME: Can we reference these instead of copying them, e.g. by passing in a reference to
+        // the `CompiledTrace` and `gidx` or better a reference to `DeoptInfo`?
+        let deoptinfo = &self.deoptinfo[&usize::from(gidx)];
+        let lives = deoptinfo
             .live_vars
             .iter()
-            .map(|(iid, _)| iid.clone())
+            .map(|(a, l)| (a.clone(), l.into()))
             .collect();
-        let callframes = self.deoptinfo[&usize::from(gidx)].inlined_frames.clone();
+        let callframes = deoptinfo.inlined_frames.clone();
+
+        // Get the root trace from the HotLocation
+        let hlarc = self.hl.upgrade().unwrap();
+        let hl = hlarc.lock();
+        let root_ctr = match &hl.kind {
+            HotLocationKind::Compiled(root_ctr) => Arc::clone(root_ctr)
+                .as_any()
+                .downcast::<X64CompiledTrace>()
+                .unwrap(),
+            // Since we are currently trying to compile a side-trace, the current HotLocationKind
+            // must be that of the root trace.
+            HotLocationKind::SideTracing { root_ctr, .. } => Arc::clone(root_ctr)
+                .as_any()
+                .downcast::<X64CompiledTrace>()
+                .unwrap(),
+            _ => panic!("Unexpected HotLocationKind"),
+        };
+
         Arc::new(YkSideTraceInfo {
-            aotlives,
+            bid: deoptinfo.bid.clone(),
+            lives,
             callframes,
+            // FIXME: This needs to point to the root trace, not the immediate parent.
+            parent: ParentTracePtr(root_ctr.entry()), //ParentTracePtr(self.entry()),
```

Comment:
Commented code.

Also this `FIXME` sounds unusually bad?!

Reply:
The comment is old. I just need to remove it.

Reply:
[868cf76](https://github.com/ykjit/yk/pull/1422/commits/868cf769fe2bf7de7725d55f99422731ca074d4c)